libftl/ingest.c

```diff
@@ -92,6 +92,96 @@ OS_THREAD_ROUTINE _ingest_get_rtt(void *data) {
     return 0;
 }
 
+ftl_status_t find_closest_available_ingest(const char* ingestHosts[], int ingestsCount, char* bestIngestHostComputed)
+{
+    ftl_ingest_t* ingestElements;
+    int i;
+
+    if ((ingestElements = malloc(sizeof(ftl_ingest_t) * ingestsCount)) == NULL) {
+        return FTL_MALLOC_FAILURE;
+    }
+
+    for (i = 0; i < ingestsCount; i++) {
+        size_t host_len = strlen(ingestHosts[i]);
+        if ((ingestElements[i].hostname = malloc(host_len)) == NULL) {
+            free(ingestElements);
+            return FTL_MALLOC_FAILURE;
+        }
+        strcpy_s(ingestElements[i].hostname, host_len, ingestHosts[i]);

```

Comment:
Is this memory copy needed? Can't you just set the pointer to the incoming pointer?

Reply:
I prefer do a mem copy here since the client should not be responsible to release a pointer that he never allocated.